src/modules/fundingManager/depositVault/interfaces/IFM_DepositVault_v1.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+interface IFM_DepositVault_v1 {
+    //--------------------------------------------------------------------------
+    // Events
+
+    /// @notice Event emitted when a deposit takes place.
+    /// @param _from The address depositing tokens.
+    /// @param _amount The amount of tokens deposited.
+    event Deposit(address indexed _from, uint _amount);
+
+    //--------------------------------------------------------------------------
+    // Functions
+
+    /// @notice Deposits a specified amount of tokens into the contract from the sender's account.
+    /// @dev    Reverts if attempting self-deposits
```

Comment:
The implementation does not revert on self deposit

Reply:
Discussed above, see [here](https://github.com/InverterNetwork/contracts/pull/626#discussion_r1707911078).

Reply:
Comment itself removed from the `deposit` function, `depositFor` is removed altogether, so this one should be done as well.